server/connectors/api-connector-jdbc/src/main/scala/com/prisma/api/connector/jdbc/database/FilterConditionBuilder.scala

```diff
@@ -101,15 +111,31 @@ trait FilterConditionBuilder extends BuilderBase {
           .select(relationColumn(relationField))
           .from(relationTable(relation))
           .where(condition.invert(invertConditionOfSubSelect))
-
+      case nested: ScalarFilter =>
+        if (nested.field.isId) {
+          val condition = nested.condition match {
+            case Equals(NullGCValue) => field(relationColumn(relationField.relatedField)).isNull
```

Comment:
Every check for `NullGCValue` is broken, we rather need to check for row existence